OptimizelySDK.Tests/OptimizelyJsonTest.cs

```diff
@@ -0,0 +1,129 @@
+
+            Assert.IsNotNull(OptimizelyJSONUsingMap);
+            Assert.IsNotNull(OptimizelyJSONUsingString);
+        }
+        [Test]
+        public void TestToStringReturnValidString()
+        {
+            Dictionary<string, object> map = new Dictionary<string, object>() {
+                { "strField", "john doe" },
+                { "intField", 12 },
+                { "objectField", new Dictionary<string, object> () {
+                        { "inner_field_int", 3 }
+                    }
+                }
+            };
+            OptimizelyJson OptimizelyJSONUsingMap = new OptimizelyJson(map, ErrorHandlerMock.Object, LoggerMock.Object);
+            string str = OptimizelyJSONUsingMap.ToString();
+            string expectedStringObj = "{\"strField\":\"john doe\",\"intField\":12,\"objectField\":{\"inner_field_int\":3}}";
+            Assert.AreEqual(expectedStringObj, str);
+        }
+
+        [Test]
+        public void TestGettingErrorUponInvalidJsonString()
+        {
+            OptimizelyJson OptimizelyJSONUsingString = new OptimizelyJson("{\"invalid\":}", ErrorHandlerMock.Object, LoggerMock.Object);
+            LoggerMock.Verify(log => log.Log(LogLevel.ERROR, "Provided string could not be converted to map."), Times.Once);
+        }
+
+        [Test]
+        public void TestGettingErrorUponNotFindingValuePath()
+        {
+            OptimizelyJson OptimizelyJSONUsingString = new OptimizelyJson("{\"invalid\":}", ErrorHandlerMock.Object, LoggerMock.Object);
+            LoggerMock.Verify(log => log.Log(LogLevel.ERROR, "Provided string could not be converted to map."), Times.Once);
+        }
+
+        [Test]
+        public void TestOptimizelyJsonGetVariablesWhenSetUsingMap()
+        {
+            OptimizelyJson OptimizelyJSONUsingMap = new OptimizelyJson(Map, ErrorHandlerMock.Object, LoggerMock.Object);
+
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<string>("strField"), "john doe");
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<int>("intField"), 12);
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<double>("doubleField"), 2.23);
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<bool>("boolField"), true);
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<int>("objectField.inner_field_int"), 3);
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<double>("objectField.inner_field_double"), 13.21);
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<string>("objectField.inner_field_string"), "john");
+            Assert.AreEqual(OptimizelyJSONUsingMap.GetValue<bool>("objectField.inner_field_boolean"), true);
+            Assert.IsTrue(OptimizelyJSONUsingMap.GetValue<Dictionary<string, object>>("objectField") is Dictionary<string, object>);
+        }
+
+        [Test]
+        public void TestOptimizelyJsonGetVariablesWhenSetUsingString()
```

Comment:
This is useless testcase.

Reply:
No, this is just to verify that all primitive type of values are getting parsed